contracts/AccountRegistration.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract AccountRegistration is ERC721Enumerable, Ownable {
+    constructor(
+        string memory name_,
+        string memory symbol_
+    ) ERC721(name_, symbol_) {}
+
+    function register(
+        address registrationAccount,
+        address tokenRecipient
+    ) external {
+        require(
+            registrationAccount == _msgSender(),
+            "AccountRegistration: registration account must be the sender"
+        );
+        uint256 tokenId = uint256(uint160(_msgSender()));
```

Comment:
Doesn this restrict us to one nft token per address? I am not sure that is the intent.

```
Oprah
- wants to give car nft to Ian (success) 
- wants to give car nft to Justin (fail bcos tokenId exists)
```

Reply:
Oh. So I recall <protocol convo> that we should get the user to do something to claim an nft. I see "AccountRegistration" does fit the use-case.

Reply:
Here the main use case is for staking. 

* User needs to do something to tell us which account they want to hold their staked funds - they can register their own address
* User can transfer this NFT wherever they want - which will be where they can receive their rewards

We can then just create claims like this for staked funds in account 0x1234

`Claim(..., caller=NftOwner(0x1234), TransferERC20(amount))`

Without having to track where the NFT is. My expectation is this contract would be deployed for staking and for other use cases where we need people to register for something it would be deployed again.

If you wanted to transfer some other NFT to users you wouldn't have to use this contract at all, you could have arbitrary NFTs in your safe and distribute them to whoever you want.

Possible changes here:

Changing the name to be clearer / docs

Depends if we want one NFT contract to deal with multiple uses like revenue distribution without redeploying (add a salt and a bit more tracking of token IDs to original account).

Reply:
Ok fair enough. So we can treat this contract to serve for this particular use-case of staking.

> Changing the name to be clearer / docs

Think this is fine

> Depends if we want one NFT contract to deal with multiple uses like revenue distribution without redeploying (add a salt and a bit more tracking of token IDs to original account).

I guess that is a big question for our nft management. Seems right to isolate this nft contract for staking and if we develop an opinion on how to organise our nfts then we can use migrate the state - possibly asking the user to re-register or (if we are generous) distribute to them.